### PR TITLE
Feedback

### PR DESCRIPTION
**Pull request by jedbrown**
:wave:! GitHub Classroom created this pull request as a place for your teacher to leave feedback on your work. It will update automatically. **Don’t close or merge this pull request**, unless you’re instructed to do so by your teacher.

In this pull request, your teacher can leave comments and feedback on your code. Click the **Subscribe** button to be notified if that happens.

Click the **Files changed** or **Commits** tab to see all of the changes pushed to `master` since the assignment started. Your teacher can see this too.

<details>
<summary>
<strong>Notes for teachers</strong>
</summary>

Use this PR to leave feedback. Here are some tips:

- Click the **Files changed** tab to see all of the changes pushed to `master` since the assignment started. To leave comments on specific lines of code, put your cursor over a line of code and click the blue **+** (plus sign). To learn more about comments, read “[Commenting on a pull request](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/commenting-on-a-pull-request)”.
- Click the **Commits** tab to see the commits pushed to `master`. Click a commit to see specific changes.
- If you turned on autograding, then click the **Checks** tab to see the results.
- This page is an overview. It shows commits, line comments, and general comments. You can leave a general comment below.

For more information about this pull request, read “[Leaving assignment feedback in GitHub](/help/leaving-feedback-in-github)”.
</details>

### CODE REVIEW

**Comment by ghost** [comment]
I like that you all ran different algorithms to try and predict housing prices, but based on your analysis I wonder if they were overfitting? Specifically:

> The code contained data for housing prices in Amsterdam. We ran it to verify this data, then ran it again for housing data from across the United States. We found that the fit scores were, in general, much lower for this new data set than for the original one. Additionally, the score ordering changed to a large degree.

In all sets the Amsterdam (train data) score was fairly high but the USA score (your test data) was low. Also for some algorithms like XGBoost you might want to do a grid search on the hyperparameters to find a more optimal fit. 

Anyways I like the graph comparisons but would like to see the code used to generate them as that appears to be missing.

edit: One more note is that you can add more parameters to your input data, but the only way to really know if your predictions are better is to do more cross-validation. Again if the goal is to generalize to houses across the US based on only data from Amsterdam, I think that a lot of the parameters will have to go to zero in order to generalize (but you would have to play with the models again to really see).

**Comment by pokemm** [comment]
Very interesting and applicable project. Like EvanRam also said, I'd like to see the code used to generate the graph comparisons. It might be interesting to also compare the runtime of each method.

**Comment by Suyichuan98** [comment]
Very cool topic. I would like to see how the prediction fits with the real data, and how effective this method is.

**Comment by techLord-pm** [comment]
I found the real world application of information pertaining to the likes of real estate, stocks, or commodities to be fascinating. One possible visual improvement could be to move the ordering rank to the title line so that it is slightly more clear.

**Comment by AhotoO** [comment]
I found this to be a very relevant topic to use regression to handle housing data. I would have liked to see a sample of both data sets to possibly give us a way to see how each set was modeled by each regressive approach. Another possible method would be to use a small set of US zip codes as the training set to compare the rest of the United States to. This could help reduce the effect of over fitting but may loss the distinctions of each method.

**Comment by JakeSwartwout** [comment]
Having built something to predict prices myself, this is a super cool idea. Great job on finding so many different packages to compare, it's super interesting to see how everyone has their own methods with different results.
Like you said, I would also be interested in trying other types of models, as I wonder if certain parameters are going to be quadratic, logarithmic, exponential, etc, in relation to the housing price.

**Comment by ghost** [comment]
It's interesting to see the different applications that regression can be used for and the comparison between different regression methods is very useful. It's fascinating how fewer tuning parameters leads to the large difference between ranking 1st in Amsterdam and ranking 8th in USA. It just shows how different regression methods can be applicable for many different purposes.

**Comment by lbakken** [comment]
I looked at an earlier version of this paper during the individual projects and liked the idea, but wanted more examples. I was glad to see the use of multiple applications and models in this final version of the project!

**Comment by harshdesh7** [comment]
For the Neural Network MLP regression, what type of model did you guys set up? Did you do any Principle Component Analysis beforehand to get rid of certain features? Also, what precautions did you take to prevent overfitting?

**Comment by DorotheaF** [comment]
I have a friend who tried to do this for the stock market and it didn't really work. I'll show him this.
I wonder why the US data scores so much lower. Also, I really appreciated your explanations of all the methods, and how direct and clear the formatting is.

**Comment by tish1781** [comment]
Great works on regression methods! I've learned a lot in the comparsion and testing between different methods. I'm also wondering if anyone has idea on the last open question?

**Comment by akessenich** [comment]
Great comparison of methods, and really interesting results. I think it was an interesting application of regression as well. I wonder if the methods were at all influenced by which dataset they were tuned on, if the datasets are different enough could that impact the effectiveness of the methods?

**Comment by MusaabFaozi** [comment]
It is very interesting project idea. I have some questions. When you say method X performs the better or worse, what do you take into consideration? Prediction accuracy, speed of the method, or a combination of both?
